khard/config.py

```diff
@@ -137,6 +139,48 @@ def _validate(config: configobj.ConfigObj) -> configobj.ConfigObj:
             raise ConfigError
         return config
 
+    @classmethod
+    def _unfold_discover_books(cls, config: configobj.ConfigObj) -> configobj.ConfigObj:
+        """Expand globs in path of addressbooks of type "discover"
+
+        This expands all addressbooks of type "discover" into (potentially)
+        multiple addressbooks of type "vdir". The names are automatically generated
+        based on the directory name.
+
+        :param config: the configuration to be changed
+        :returns: the changed configuration with no "discover" addressbooks
+        """
+        for section_name, book in config["addressbooks"].copy().items():
+            if book["type"] != "discover":
+                continue
+            hits = iglob(os.path.expanduser(book["path"]), recursive=True)
+            dirs = cls._find_leaf_dirs(hits)
+            for bookpath in dirs:
+                bookname = os.path.basename(bookpath)
+                # Make sure our name is unique
+                while bookname in config["addressbooks"]:
+                    bookname += "-1"
```

Comment:
Can you please do this with a counter.  Currently this results in names like `foo`, `foo-1`, `foo-1-1` but I would prefer `foo`, `foo-1`, `foo-2`, ...

```suggestion
                i = 0
                while bookname in config["addressbooks"]:
                    i += 1
                    bookname += f"-{i}"
```